data_encryption/models/encrypted_data.py

```diff
@@ -0,0 +1,130 @@
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+import logging
+import json
+
+from odoo import api, fields, models, SUPERUSER_ID
+from odoo.exceptions import AccessError, ValidationError
+from odoo.tools.config import config
+from odoo.tools.translate import _
+from odoo.tools import ormcache
+
+_logger = logging.getLogger(__name__)
+
+try:
+    from cryptography.fernet import Fernet, InvalidToken
+except ImportError as err:  # pragma: no cover
+    _logger.debug(err)
+
+
+class EncryptedData(models.Model):
+    """Model to store encrypted data by environment (prod, preprod...)"""
+
+    _name = 'encrypted.data'
+    _description = 'Store any encrypted data by environment'
+
+    name = fields.Char(
+        required=True, readonly=True, index=True,
+        help="Technical name")
+    environment = fields.Char(
+        required=True,
+        index=True,
+        help="Concerned Odoo environment (prod, preprod...)")
+    encrypted_data = fields.Binary()
+
+    _sql_constraints = [
+        ('name_environment_uniq', 'unique (name, environment)',
+         'You can not store multiple encrypted data for the same record and \
+          environment')
+    ]
+
+    @api.model
+    @ormcache('self._uid', 'name', 'env')
+    def _get(self, name, env=None):
+        if self.env.context.get('bin_size'):
+            self = self.with_context(bin_size=False)
+        if self.env.user.id != SUPERUSER_ID:
```

Comment:
better to use `self.env.user._is_superuser()`

Reply:
Will do